haskell/src/Stablecoin/Client/Parser.hs

```diff
@@ -0,0 +1,683 @@
+  { gmaoMinter :: AddressOrAlias
+  }
+  deriving stock Show
+
+clientArgsParser :: Opt.Parser ClientArgs
+clientArgsParser =
+  ClientArgs
+    <$> clientConfigParser (pure Nothing)
+    <*> globalOptionsParser
+    <*> clientArgsRawParser
+
+globalOptionsParser :: Opt.Parser GlobalOptions
+globalOptionsParser =
+  GlobalOptions
+    <$> addressOrAliasOption
+      (Just $ AddressAlias "stablecoin-user")
+      (#name .! "user")
+      (#help .! "User to send operations as")
+    <*> addressOrAliasOption
+      (Just $ AddressAlias "stablecoin")
+      (#name .! "contract")
+      (#help .! "The stablecoin contract address/alias to use")
+
+clientArgsRawParser :: Opt.Parser ClientArgsRaw
+clientArgsRawParser = Opt.subparser $
+  deployCmd
+  <> transferCmd
+  <> getBalanceOfCmd
+  <> updateOperatorsCmd
+  <> isOperatorCmd
+  <> permissionsDescriptorCmd
+  <> pauseCmd
+  <> unpauseCmd
+  <> configureMinterCmd
+  <> removeMinterCmd
+  <> mintCmd
+  <> burnCmd
+  <> transferOwnershipCmd
+  <> acceptOwnershipCmd
+  <> changeMasterMinterCmd
+  <> changePauserCmd
+  <> setTransferlistCmd
+  <> getBalanceCmd
+  <> getPausedCmd
+  <> getContractOwnerCmd
+  <> getPendingContractOwnerCmd
+  <> getMasterMinterCmd
+  <> getPauserCmd
+  <> getTransferlistCmd
+  <> getTotalSupplyCmd
+  <> getMintingAllowanceCmd
+  <> getTokenMetadataCmd
+  where
+    deployCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    deployCmd =
+      mkCommandParser
+        "deploy"
+        (CmdDeployContract <$> deployContractOptions)
+        "Deploy Stablecoin contract to the chain. An alias 'stablecoin' will be created."
+
+    deployContractOptions :: Opt.Parser DeployContractOptions
+    deployContractOptions = do
+      dcoMasterMinter <-
+        addressOrAliasOption
+          Nothing
+          (#name .! "master-minter")
+          (#help .! "Address or alias of the 'Master Minter' role")
+      dcoContractOwner <-
+        addressOrAliasOption
+          Nothing
+          (#name .! "contract-owner")
+          (#help .! "Address or alias of the 'Contract Owner' role")
+      dcoPauser <-
+        addressOrAliasOption
+          Nothing
+          (#name .! "pauser")
+          (#help .! "Address or alias of the 'Pauser' role")
+      dcoTransferlist <-
+        addressOrAliasOption
+          Nothing
+          (#name .! "transferlist")
+          (#help .! "Address or alias of the Transferlist contract")
+      dcoTokenSymbol <-
+        mTextOption
+          Nothing
+          (#name .! "token-symbol")
+          (#help .! "Token symbol")
+      dcoTokenName <-
+        mTextOption
+          Nothing
+          (#name .! "token-name")
+          (#help .! "Token name")
+      dcoTokenDecimals <-
+        naturalOption
+          Nothing
+          (#name .! "token-decimals")
+          (#help .! ("Number of digits to use after the decimal point " <>
+                     "when displaying the token amounts"))
```

Comment:
I think token metadata will often be out of interest, so I would define some default values for `dcoTokenSymbol`, `dcoTokenName`, `dcoTokenDecimals`.

Reply:
I suppose [these](https://github.com/tqtezos/stablecoin/blob/master/haskell/src/Lorentz/Contracts/Stablecoin.hs#L236-L238) would qualify as sensible defaults, is that right?

```haskell
      (#token_id .! 0, #mdr .! ( #symbol .! [mt|USDC|]
                               , #mdr2 .! ( #name .! [mt|USDC|]
                                          , #mdr3 .! ( #decimals .! 8, #extras .! mempty))))
```

Reply:
Yes, that's fine.

Reply:
Done